translator/jsonconfig/mergeJsonConfig.go

```diff
@@ -16,20 +17,29 @@ import (
 	"github.com/aws/amazon-cloudwatch-agent/translator/util"
 )
 
-func MergeJsonConfigMaps(jsonConfigMapMap map[string]map[string]interface{}, defaultJsonConfigMap map[string]interface{}, multiConfig string) (map[string]interface{}, error) {
+func MergeJsonConfigMaps(jsonConfigMapMap map[string]map[string]interface{}, defaultJsonConfigMap map[string]interface{}, multiConfig string, strictValidation bool, inputJsonDirPathErr error ) (map[string]interface{}, error) {
 	if jsonConfigMapMap == nil || len(jsonConfigMapMap) == 0 {
 		if os.Getenv(config.USE_DEFAULT_CONFIG) == config.USE_DEFAULT_CONFIG_TRUE {
 			// When USE_DEFAULT_CONFIG is true, ECS and EKS will be supposed to use different default config. EKS default config logic will be added when necessary
 			if ecsutil.GetECSUtilSingleton().IsECS() {
-				fmt.Println("No json config files found, use the default ecs config")
+				fmt.Println("Invalid agent's json config files, use the default ecs config")
 				return util.GetJsonMapFromJsonBytes([]byte(config.DefaultECSJsonConfig()))
 			}
 		}
-		if multiConfig == "remove" {
-			fmt.Println("No json config files found, please provide config, exit now")
-			os.Exit(config.ERR_CODE_NOJSONFILE)
+
+		if errors.Is(inputJsonDirPathErr, os.ErrNotExist) {
+			//When there is no input from flag --input, --input-dir and cannot find agent's config through containerized environment
+			log.Printf("No agent's json config was found from containerized environment or file path.")
+			if strictValidation{
+				os.Exit(config.ERR_CODE_NOJSONFILE)
+			}
+			log.Printf("Use the default agent's json config.")
```

Comment:
This feels like a bad code smell to me. I could understand returning from a function early, but having a log/other actions afterwards if the condition isn't met, but I don't feel like that same logic applies to this scenario where we abruptly terminate the process with an exit call. 

In this code, we check if the error is a file/dir doesn't exist, and then if there is strict validation, we terminate. In the `else if`, if there is strict validation or we want to remove configs, then we log exit. Otherwise, we fail through. 

I haven't figured out how to rework this, but I will think about this more.

Reply:
Personally, I would agree on logs/action afterwards actions afterwards would not be a best practice. Cause the whole reason I writing this one was to simplifying the code and manage all error states in `MergeJsonConfigMaps`. However, if that's not what we desired, then keeping an original code would be fine by me. Cause the question I only keep question is: how was we able to differentiate between `Invalid Config` and `NoConfig` through the function `MergeJsonConfigMaps`. If its only serves as a helper one then only `logging Invalid`  would be a better idea for me.